ovirt-imageio.spec.in

```diff
@@ -32,10 +56,24 @@ install -D -m 0755 --directory %{buildroot}%{admin_confdir}/conf.d
 install -D -m 0644 data/README %{buildroot}%{admin_confdir}
 install -D -m 0644 data/%{name}.service %{buildroot}%{_unitdir}/%{name}.service
 
+%if %{with_python38}
+# Python 3.8 wins shebangs mangling, but we want daemon to be available only
+# for python 3, so we need to override
+mv %{buildroot}%{_bindir}/%{name} %{buildroot}%{_bindir}/%{name}.bck
+mv %{buildroot}%{_bindir}/%{name}ctl %{buildroot}%{_bindir}/%{name}ctl.bck
+sed -e 's|python3.8|python3|g' %{buildroot}%{_bindir}/%{name}.bck > %{buildroot}%{_bindir}/%{name}
+sed -e 's|python3.8|python3|g' %{buildroot}%{_bindir}/%{name}ctl.bck > %{buildroot}%{_bindir}/%{name}ctl
+rm %{buildroot}%{_bindir}/%{name}.bck
+rm %{buildroot}%{_bindir}/%{name}ctl.bck
```

Comment:
This is horrible hack :-)

Why not use sed -i to change the file inplace?

Reply:
Yes, it is. The problem is that shebang mangling is performed for executable files, so we need fix shebang here, then make the file nonexecutable and then only during RPM installation make it executable again (line 136-137).

But Marcin found out better solution, I will post it soon

Reply:
Not needed anymore, Marcin's solution is better and works fine